runtime/autoload/health/provider.vim

```diff
@@ -244,39 +217,31 @@ function! s:version_info(python) abort
   let nvim_path_base = fnamemodify(nvim_path, ':~:h')
   let version_status = 'unknown; '.nvim_path_base
   if !s:is_bad_response(nvim_version) && !s:is_bad_response(pypi_version)
-    if s:version_cmp(nvim_version, pypi_version) == -1
+    if luaeval('vim.cmp_versions(_A[1], _A[2])', [nvim_version, pypi_version]) == -1
       let version_status = 'outdated; from '.nvim_path_base
     else
       let version_status = 'up to date'
     endif
   endif
 
-  return [python_version, nvim_version, pypi_version, version_status]
-endfunction
-
-" Check the Python interpreter's usability.
-function! s:check_bin(bin) abort
-  if !filereadable(a:bin) && (!has('win32') || !filereadable(a:bin.'.exe'))
-    call health#report_error(printf('"%s" was not found.', a:bin))
-    return 0
-  elseif executable(a:bin) != 1
-    call health#report_error(printf('"%s" is not executable.', a:bin))
-    return 0
-  endif
-  return 1
+  return [nvim_version, pypi_version, version_status]
 endfunction
 
 " Check "loaded" var for given a:provider.
 " Returns 1 if the caller should return (skip checks).
 function! s:disabled_via_loaded_var(provider) abort
   let loaded_var = 'g:loaded_'.a:provider.'_provider'
-  if exists(loaded_var) && !exists('*provider#'.a:provider.'#Call')
+  let call_func = 'provider#'.a:provider.'#Call'
+  if exists(loaded_var) && !exists('*'.call_func)
     let v = eval(loaded_var)
     if 0 is v
       call health#report_info('Disabled ('.loaded_var.'='.v.').')
       return 1
     else
-      call health#report_info('Disabled ('.loaded_var.'='.v.').  This might be due to some previous error.')
+      call health#report_info(printf('%s set to %s but %s function does not exist.  '
+        \ . 'This might be due to some previous error.  If you are setting %s manually '
+        \ . 'to disable the %s provider, set it to 0 to avoid this message.',
+        \ loaded_var, v, call_func, loaded_var, a:provider))
```

Comment:
I thought the original message was somewhat confusing/unclear when reading the code -- the function is called `s:disabled_via_loaded_var`, the previous branch reports "Disabled" and returns 1, this one *also* reports "Disabled" but doesn't.